apps/designer/app/designer/features/props-panel/control.tsx

```diff
@@ -142,84 +220,150 @@ const RangeControl = ({
     </Flex>
   </Flex>
 );
+*/
 
 const NotImplemented = () => <div />;
 
-type PrimitiveControlProps = BaseControlProps & {
-  type: "array" | "boolean" | "date" | "number" | "object" | "text";
+const includes = <T extends string>(arr: readonly T[], v: string): v is T => {
+  return arr.includes(v as never);
 };
 
-type ColorControlProps = BaseControlProps & {
-  type: "color";
-  presetColors?: Array<string>;
+const isStringArray = (arr: unknown): arr is Array<string> => {
+  return Array.isArray(arr) && arr.every((v) => typeof v === "string");
 };
 
-type FileControlProps = BaseControlProps & {
-  type: "file";
-  accept: string;
+type ControlProps = {
+  component: Instance["component"];
+  userProp: UserProp;
+  onChangePropValue: (value: string | boolean) => void;
+  onChangePropAsset: (asset: Asset) => void;
 };
 
-type RangeControlProps = BaseControlProps<number> & {
-  type: "range";
-  min: number;
-  max: number;
-  step: number;
-};
+// eslint-disable-next-line func-style
+export function Control({
+  component,
+  userProp,
+  onChangePropValue,
+  onChangePropAsset,
+}: ControlProps) {
+  const meta = componentsMeta[component];
+  const argType = meta.argTypes?.[userProp.prop as keyof typeof meta.argTypes];
 
-type OptionsControlProps = BaseControlProps & {
-  type:
-    | "radio"
-    | "inline-radio"
-    | "check"
-    | "inline-check"
-    | "select"
-    | "multi-select";
-  options: Array<string>;
-};
+  const defaultValue = argType?.control.defaultValue;
+  const type = argType?.control.type;
+
+  if (type == null) {
```

Comment:
can type be null and undefined? if yes then its correct

Reply:
it can, its even `any` as control is not defined on argtype